### DISCUSSION TITLE
Obtaining official Einstein A20 SDK

### DISCUSSION THREAD

**Discussion by loredous** [mdubinko/gsoc-contrib#12] [Ideas]
Unfortunately at this point in time, it looks like the "official" Cubietech download site doesn't have the full Einstein A20 SDK available. I have put out feelers through E-mail to Cubietech, but I don't have high hopes at this point. Ideally, they would be able to provide a full Buildroot configuration that we can base new images off of, but we'll see what we can get. 

If anyone has a line on where we might get information like this, that would be fantastic.

**Comment by 2E0WKS** [comment]
Shouldn't need it as functional debian running on sd card thanks to others helpful contributions-I suspect unofficial push in right direction as it is beta functional. 
8gb internal waste of time and still boots from SD card if you want. 

The trick would be to see what uboot uncovers on boot as I believe this sets up WiFi to gpio via uboot software configuration.
As it stands I can run gsoc off internal 1.3 image or boot to debian and apt is installed so apt-get install xyz should be possible-Once I have WiFi via a dongle and firmware 
I plan on
Adding
Auto login
Synaptic package manager
Adding shortcut to gsoc app
Adding other software.
Creating an image of final sd card no issue

**Reply by d3cker** [reply]
Hey. 
Take a look here: https://github.com/d3cker/gsoc-puhumod . We made a script that creates Debian image for GSOC based on official firmware kernel. I was able to run WSJT-X , Xiegu APP etc, boot from USB stick. In general we did some progress, yet making full, free and open source kernel for that device was a dead end for me. I remember I did boot custom built kernel, but I was not able to make video device up. With official image kernel, there is a limited number of devices that are supported (wifi dongles etc). Once I noticed that Xiegu dropped support for GSOC I decided to leave it for a certain death. The only use case I found, was a possibility to sniff Xiegu protocol for G90 head panel : https://github.com/d3cker/g90puhuApp

Regards, 
d3cker (SP6XD)

**Comment by 2E0WKS** [comment]
d3cker (SP6XD) thank you for the hints, tips and fine effort with scripts. 
As I understand it this is a cubieboard2 / Einstein board and I understand the kernel module issues you describe won't load modules, I don't however quite understand what stops the switch to other Linux or pulling via apt get from sources? 
Is Uboot or Kernel security CRC or similar secure boot stopping this?
I did investigate Armbian Cubieboard2 and it seems a dead end, also livesuite A20 flash tool- however not keen to destroy a no warranty second hand GSOC.
Thank you for all your efforts

**Comment by 2E0WKS** [comment]
I have Armbian jammy debian running but not to desktop, I believe it is Lcd settings as have prompt.
Sure sleeping on it will help.
